src/test/java/io/confluent/examples/streams/IntegrationTestUtils.java

```diff
@@ -397,21 +398,43 @@ private static boolean continueConsuming(final int messagesConsumed, final int m
                                                    final Serializer<V> valueSerializer) {
     for (final KeyValue<K, V> entity : records) {
       final ConsumerRecord<byte[], byte[]> consumerRecord = new ConsumerRecord<>(
-        topic,
-        0,
-        0,
-        0,
-        TimestampType.CREATE_TIME,
-        ConsumerRecord.NULL_CHECKSUM,
-        ConsumerRecord.NULL_SIZE,
-        ConsumerRecord.NULL_SIZE,
-        keySerializer.serialize(topic, entity.key),
-        valueSerializer.serialize(topic, entity.value)
+          topic,
+          0,
+          0,
+          0,
+          TimestampType.CREATE_TIME,
+          ConsumerRecord.NULL_CHECKSUM,
+          ConsumerRecord.NULL_SIZE,
+          ConsumerRecord.NULL_SIZE,
+          keySerializer.serialize(topic, entity.key),
+          valueSerializer.serialize(topic, entity.value)
       );
       topologyTestDriver.pipeInput(consumerRecord);
     }
   }
 
+  /**
+   * Like {@link IntegrationTestUtils#produceValuesSynchronously(String, Collection, Properties)}, except for use with
+   * TopologyTestDriver tests, rather than "native" Kafka broker tests.
+   *
+   * @param topic              Kafka topic to write the data records to
+   * @param values             Message values to write to Kafka (keys will have type byte[] and be set to null)
+   * @param topologyTestDriver The {@link TopologyTestDriver} to send the data records to
+   * @param valueSerializer    The {@link Serializer} corresponding to the value type
+   * @param <V>                Value type of the data records
+   */
+  static <K, V> void produceValuesSynchronously(final String topic,
```

Comment:
This is a nice addition, but can you target this change to `4.1.0-post`? Otherwise, every change to every test is going to run into merge conflicts when they hit this branch.

Reply:
Note, offhand, it seems like this whole PR could be targeted to `4.1.0-post`

Reply:
> This is a nice addition, but can you target this change to 4.1.0-post? Otherwise, every change to every test is going to run into merge conflicts when they hit this branch.

Why would that be so?  The changes in this PR are targeting `5.2.0`, and would therefore not be pushed into any `4.x` branches?

Also, I tried rebasing this PR on `4.1.0-post` for the fun of it, but there are 120+ commits, many of which result into merge failures.  I don't think we need to retroactively also change the `4.x` ones.

Reply:
That makes sense. In that case, I would recommend reverting the "cleanup" portions of this PR... The main problem is that every time I have to make a change to the examples in an older branch, I already spend hours resolving merge conflicts. If we're going to make cross-cutting changes, they really need to be done in the first applicable branch.

On the other hand, your main goal here is to add some new examples, which is not cross-cutting, and I'd have no problem just targeting 5.2.0 for that.